### PR TITLE
Ensure address is correct

### PR DESCRIPTION
**Pull request by phohenberger**
Tokens containing `ws` were getting also replaced. This only replaces the protocol.

### CODE REVIEW

File: zndraw/zndraw.py [hunk 1]
```diff
@@ -196,9 +197,9 @@ def __delitem__(self, index: int | slice | list[int]):
     def _repr_html_(self):
         from IPython.display import IFrame
 
-        address = f"{self.url}/token/{self.token}"
         # TODO: save address and do not replace in post_init
-        address = address.replace("ws", "http")
+        parsed_url = urllib.parse.urlparse(f"{self.url}/token/{self.token}")
+        address = parsed_url._replace(scheme="http").geturl()
```
Comment (PythonFZ) [inline zndraw/zndraw.py:202]: Is there a way to rename the schema without using the "private" `_replace` method?

**Review by PythonFZ — CHANGES_REQUESTED** [review]
Can you move the function to utils, add a test case with / without `ws` in the token and also use it https://github.com/zincware/ZnDraw/blob/cb3631d5eda77a9e2264cdddba974bf08e2bd301/zndraw/zndraw.py#L85

**Comment by PythonFZ** [comment]
I have added a test for https (maybe try it with the hosted zndraw). I think this should be fixed.

**Comment by phohenberger** [comment]
The previous replacement was also only for `http`. The function would have to know what protocol it should be using, based on the previous code I just assumed it would be `http`. I do not know when it should have to `https` or `http`

**Comment by PythonFZ** [comment]
> The previous replacement was also only for `http`. The function would have to know what protocol it should be using, based on the previous code I just assumed it would be `http`. I do not know when it should have to `https` or `http`

When the adress is given as `https` it should be `wss` and if given as `http` the protocol is `ws`. The `replace` cmd has worked for both, because it only replaced the first part of `http_s_` with `ws`. If you say it is not required when using e.g.

```python
from zndraw import ZnDraw

vis = ZnDraw(url="https://zndraw.icp.uni-stuttgart.de", token="erik_e")
```

We can remove the tests but I'd assume it needs the correct protocol?

**Comment by phohenberger** [comment]
Ah, alright. No, i can just filter the previous `scheme` and use `https` for `wss` and ``http` for `ws`. This should work

**Review by PythonFZ — APPROVED** [review]

**Pull request merged**